### PR TITLE
Add favagtk

### PR DESCRIPTION
**Pull request by johannesjh**
Thank you for considering this pull request. I would like to publish [favagtk](https://gitlab.gnome.org/johannesjh/favagtk) on flathub. This is a GNOME/GTK4 app that wraps fava, an open-source plaintext accounting software, in order to make it easy to use fava on the GNOME desktop.

### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. 
  - I have developed [favgtk](https://gitlab.gnome.org/johannesjh/favagtk) and so far, I am the sole author
  - favagtk is a thin wrapper around fava, and I have contacted fava's maintainers to ask for their consent, see https://github.com/beancount/fava/issues/1209
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by hfiguiere** [comment]
bot, build org.gnome.gitlab.johannesjh.favagtk

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: org.gnome.gitlab.johannesjh.favagtk.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    "/share/doc",
+    "/share/gtk-doc",
+    "/share/man",
+    "/share/pkgconfig",
+    "*.la",
+    "*.a"
+  ],
+  "modules": [
+    "requirements/webkit.json",
+    "requirements/blueprint-compiler.json",
```
[the remark this answers is not in the document]
Comment (johannesjh) [reply org.gnome.gitlab.johannesjh.favagtk.json:27]: hi hubert, thank you for your comment. are the separate json files a problem? / why would they be a problem? sorry, it just is not obvious to me. note, another app also references a separate webkit.json file, see https://github.com/flathub/org.gabmus.gfeeds, and flathub's shared module use the same programming pattern of splitting the manifest into multiple files. thank you!

File: requirements/python3-requirements.json [hunk 2]
```diff
@@ -0,0 +1,953 @@
...
+          "sha256": "939de3e7a6161af0c887ef91b7d41a53e7c5a1ca976325f429cb46ea9bc30ecc"
+        },
+        {
+          "type": "file",
+          "url": "https://files.pythonhosted.org/packages/19/c7/fa589626997dd07bd87d9269342ccb74b1720384a4d739a1872bd84fbe68/chardet-4.0.0-py2.py3-none-any.whl",
+          "sha256": "f864054d66fd9118f2e67044ac8981a54775ec5b67aed0441892edb553d21da5"
+        },
+        {
+          "type": "file",
+          "url": "https://files.pythonhosted.org/packages/ce/69/66c2112761ed295a8469c3edf9f68b2f37c878a95fd715cab862d1ec92a1/lxml-4.8.0-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.manylinux_2_24_x86_64.whl",
```
[the remark this answers is not in the document]
Comment (johannesjh) [reply requirements/python3-requirements.json:439]: hi, what would you recommend to do? Would a build purely for x86 also be acceptable?

[the remark this answers is not in the document]
Comment (hfiguiere) [reply requirements/python3-requirements.json:439]: Given the structure of the python specific manifest, I was assuming you knew about https://github.com/flatpak/flatpak-builder-tools/tree/master/pip to generate the dependencies. But given the binary wheels, you didn't.

Usually it's either an `any` wheel, or a source packages.

File: org.gnome.gitlab.johannesjh.favagtk.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (hfiguiere) [reply org.gnome.gitlab.johannesjh.favagtk.json:27]: It is not a recommended practice. It's harder to maintain, and should only be used either for very complex dependencies, for auto-generated manifest (like for python), or for the use of shared-modules.

I would recommend the same for the package you linked to.

File: requirements/python3-requirements.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (johannesjh) [reply requirements/python3-requirements.json:439]: I know about the [flatpak-pip-generator](https://github.com/flatpak/flatpak-builder-tools/blob/master/pip/flatpak-pip-generator). When I used the script, it spit out a list of python source packages that needed a lot of dependencies to compile and install, that's why I could not get the installation to work using source packages. Wheels were much easier, but they are platform specific.

Do I need to get an arch independent build to work, or would a x86 build be acceptable? 

If it must be, I can give it another try. I found that another app called Paperwork was successful in installing scikit-learn from python source packages, compare https://github.com/flathub/work.openpaper.Paperwork/blob/master/shared-modules/scikit-learn-0.24.0.json, so I could try to copy their module and see if it works for me.

**Comment by barthalion** [comment]
Build fails for aarch64:

```
FAILED: Source/WebCore/CMakeFiles/WebCore.dir/__/__/WebCore/DerivedSources/unified-sources/UnifiedSource-3a52ce78-119.cpp.o 
/usr/bin/ccache /usr/bin/c++ -DBUILDING_GTK__=1 -DBUILDING_WEBKIT -DBUILDING_WITH_CMAKE=1 -DBUILDING_WebCore -DBWRAP_EXECUTABLE=\"/app/bin/bwrap\" -DDBUS_PROXY_EXECUTABLE=\"/app/bin/xdg-dbus-proxy\" -DGETTEXT_PACKAGE=\"WebKit2GTK-5.0\" -DHAVE_CONFIG_H=1 -DJSC_GLIB_API_ENABLED -DPAS_BMALLOC=1 -DSTATICALLY_LINKED_WITH_PAL -DWEBKITGTK_API_VERSION_STRING=\"5.0\" -I/run/build/webkit2gtk-5/Source/WebCore/platform/graphics/libwpe -I/run/build/webkit2gtk-5 -I/run/build/webkit2gtk-5/WebCore/DerivedSources -I/run/build/webkit2gtk-5/Source/WebCore -I/run/build/webkit2gtk-5/Source/WebCore/Modules/WebGPU -I/run/build/webkit2gtk-5/Source/WebCore/Modules/airplay -I/run/build/webkit2gtk-5/Source/WebCore/Modules/applepay -I/run/build/webkit2gtk-5/Source/WebCore/Modules/applepay/paymentrequest -I/run/build/webkit2gtk-5/Source/WebCore/Modules/applicationmanifest -I/run/build/webkit2gtk-5/Source/WebCore/Modules/async-clipboard -I/run/build/webkit2gtk-5/Source/WebCore/Modules/beacon -I/run/build/webkit2gtk-5/Source/WebCore/Modules/cache -I/run/build/webkit2gtk-5/Source/WebCore/Modules/contact-picker -I/run/build/webkit2gtk-5/Source/WebCore/Modules/cookie-consent -I/run/build/webkit2gtk-5/Source/WebCore/Modules/credentialmanagement -I/run/build/webkit2gtk-5/Source/WebCore/Modules/encryptedmedia -I/run/build/webkit2gtk-5/Source/WebCore/Modules/encryptedmedia/legacy -I/run/build/webkit2gtk-5/Source/WebCore/Modules/entriesapi -I/run/build/webkit2gtk-5/Source/WebCore/Modules/fetch -I/run/build/webkit2gtk-5/Source/WebCore/Modules/filesystemaccess -I/run/build/webkit2gtk-5/Source/WebCore/Modules/geolocation -I/run/build/webkit2gtk-5/Source/WebCore/Modules/highlight -I/run/build/webkit2gtk-5/Source/WebCore/Modules/indexeddb -I/run/build/webkit2gtk-5/Source/WebCore/Modules/indexeddb/client -I/run/build/webkit2gtk-5/Source/WebCore/Modules/indexeddb/server -I/run/build/webkit2gtk-5/Source/WebCore/Modules/indexeddb/shared -I/run/build/webkit2gtk-5/Source/WebCore/Modules/mediacapabilities -I/run/build/webkit2gtk-5/Source/WebCore/Modules/mediacontrols -I/run/build/webkit2gtk-5/Source/WebCore/Modules/mediarecorder -I/run/build/webkit2gtk-5/Source/WebCore/Modules/mediasession -I/run/build/webkit2gtk-5/Source/WebCore/Modules/mediasource -I/run/build/webkit2gtk-5/Source/WebCore/Modules/mediastream -I/run/build/webkit2gtk-5/Source/WebCore/Modules/mediastream/libwebrtc -I/run/build/webkit2gtk-5/Source/WebCore/Modules/model-element -I/run/build/webkit2gtk-5/Source/WebCore/Modules/model-element/dummy -I/run/build/webkit2gtk-5/Source/WebCore/Modules/navigatorcontentutils -I/run/build/webkit2gtk-5/Source/WebCore/Modules/notifications -I/run/build/webkit2gtk-5/Source/WebCore/Modules/paymentrequest -I/run/build/webkit2gtk-5/Source/WebCore/Modules/permissions -I/run/build/webkit2gtk-5/Source/WebCore/Modules/pictureinpicture -I/run/build/webkit2gtk-5/Source/WebCore/Modules/plugins -I/run/build/webkit2gtk-5/Source/WebCore/Modules/push-api -I/run/build/webkit2gtk-5/Source/WebCore/Modules/remoteplayback -I/run/build/webkit2gtk-5/Source/WebCore/Modules/speech -I/run/build/webkit2gtk-5/Source/WebCore/Modules/storage -I/run/build/webkit2gtk-5/Source/WebCore/Modules/streams -I/run/build/webkit2gtk-5/Source/WebCore/Modules/web-locks -I/run/build/webkit2gtk-5/Source/WebCore/Modules/webaudio -I/run/build/webkit2gtk-5/Source/WebCore/Modules/webauthn -I/run/build/webkit2gtk-5/Source/WebCore/Modules/webauthn/cbor -I/run/build/webkit2gtk-5/Source/WebCore/Modules/webauthn/fido -I/run/build/webkit2gtk-5/Source/WebCore/Modules/webcodecs -I/run/build/webkit2gtk-5/Source/WebCore/Modules/webdatabase -I/run/build/webkit2gtk-5/Source/WebCore/Modules/webdriver -I/run/build/webkit2gtk-5/Source/WebCore/Modules/websockets -I/run/build/webkit2gtk-5/Source/WebCore/Modules/webxr -I/run/build/webkit2gtk-5/Source/WebCore/accessibility -I/run/build/webkit2gtk-5/Source/WebCore/accessibility/isolatedtree -I/run/build/webkit2gtk-5/Source/WebCore/animation -I/run/build/webkit2gtk-5/Source/WebCore/bindings -I/run/build/webkit2gtk-5/Source/WebCore/bindings/js -I/run/build/webkit2gtk-5/Source/WebCore/bridge -I/run/build/webkit2gtk-5/Source/WebCore/bridge/c -I/run/build/webkit2gtk-5/Source/WebCore/bridge/jsc -I/run/build/webkit2gtk-5/Source/WebCore/contentextensions -I/run/build/webkit2gtk-5/Source/WebCore/crypto -I/run/build/webkit2gtk-5/Source/WebCore/crypto/algorithms -I/run/build/webkit2gtk-5/Source/WebCore/crypto/keys -I/run/build/webkit2gtk-5/Source/WebCore/crypto/parameters -I/run/build/webkit2gtk-5/Source/WebCore/css -I/run/build/webkit2gtk-5/Source/WebCore/css/calc -I/run/build/webkit2gtk-5/Source/WebCore/css/parser -I/run/build/webkit2gtk-5/Source/WebCore/css/typedom -I/run/build/webkit2gtk-5/Source/WebCore/css/typedom/numeric -I/run/build/webkit2gtk-5/Source/WebCore/css/typedom/transform -I/run/build/webkit2gtk-5/Source/WebCore/cssjit -I/run/build/webkit2gtk-5/Source/WebCore/display -I/run/build/webkit2gtk-5/Source/WebCore/display/compositing -I/run/build/webkit2gtk-5/Source/WebCore/display/css -I/run/build/webkit2gtk-5/Source/WebCore/dom -I/run/build/webkit2gtk-5/Source/WebCore/dom/messageports -I/run/build/webkit2gtk-5/Source/WebCore/domjit -I/run/build/webkit2gtk-5/Source/WebCore/editing -I/run/build/webkit2gtk-5/Source/WebCore/fileapi -I/run/build/webkit2gtk-5/Source/WebCore/history -I/run/build/webkit2gtk-5/Source/WebCore/html -I/run/build/webkit2gtk-5/Source/WebCore/html/canvas -I/run/build/webkit2gtk-5/Source/WebCore/html/forms -I/run/build/webkit2gtk-5/Source/WebCore/html/parser -I/run/build/webkit2gtk-5/Source/WebCore/html/shadow -I/run/build/webkit2gtk-5/Source/WebCore/html/track -I/run/build/webkit2gtk-5/Source/WebCore/inspector -I/run/build/webkit2gtk-5/Source/WebCore/inspector/agents -I/run/build/webkit2gtk-5/Source/WebCore/inspector/agents/page -I/run/build/webkit2gtk-5/Source/WebCore/inspector/agents/worker -I/run/build/webkit2gtk-5/Source/WebCore/layout -I/run/build/webkit2gtk-5/Source/WebCore/layout/formattingContexts -I/run/build/webkit2gtk-5/Source/WebCore/layout/formattingContexts/block -I/run/build/webkit2gtk-5/Source/WebCore/layout/formattingContexts/block/tablewrapper -I/run/build/webkit2gtk-5/Source/WebCore/layout/formattingContexts/flex -I/run/build/webkit2gtk-5/Source/WebCore/layout/floats -I/run/build/webkit2gtk-5/Source/WebCore/layout/formattingContexts/inline -I/run/build/webkit2gtk-5/Source/WebCore/layout/formattingContexts/inline/display -I/run/build/webkit2gtk-5/Source/WebCore/layout/formattingContexts/inline/invalidation -I/run/build/webkit2gtk-5/Source/WebCore/layout/formattingContexts/inline/text -I/run/build/webkit2gtk-5/Source/WebCore/layout/integration -I/run/build/webkit2gtk-5/Source/WebCore/layout/layouttree -I/run/build/webkit2gtk-5/Source/WebCore/layout/formattingContexts/table -I/run/build/webkit2gtk-5/Source/WebCore/loader -I/run/build/webkit2gtk-5/Source/WebCore/loader/appcache -I/run/build/webkit2gtk-5/Source/WebCore/loader/archive -I/run/build/webkit2gtk-5/Source/WebCore/loader/archive/mhtml -I/run/build/webkit2gtk-5/Source/WebCore/loader/cache -I/run/build/webkit2gtk-5/Source/WebCore/loader/icon -I/run/build/webkit2gtk-5/Source/WebCore/mathml -I/run/build/webkit2gtk-5/Source/WebCore/page -I/run/build/webkit2gtk-5/Source/WebCore/page/csp -I/run/build/webkit2gtk-5/Source/WebCore/page/scrolling -I/run/build/webkit2gtk-5/Source/WebCore/platform -I/run/build/webkit2gtk-5/Source/WebCore/platform/animation -I/run/build/webkit2gtk-5/Source/WebCore/platform/audio -I/run/build/webkit2gtk-5/Source/WebCore/platform/calc -I/run/build/webkit2gtk-5/Source/WebCore/platform/encryptedmedia -I/run/build/webkit2gtk-5/Source/WebCore/platform/gamepad -I/run/build/webkit2gtk-5/Source/WebCore/platform/graphics -I/run/build/webkit2gtk-5/Source/WebCore/platform/graphics/cpu/arm -I/run/build/webkit2gtk-5/Source/WebCore/platform/graphics/cpu/arm/filters -I/run/build/webkit2gtk-5/Source/WebCore/platform/graphics/displaylists -I/run/build/webkit2gtk-5/Source/WebCore/platform/graphics/filters -I/run/build/webkit2gtk-5/Source/WebCore/platform/graphics/filters/software -I/run/build/webkit2gtk-5/Source/WebCore/platform/graphics/iso -I/run/build/webkit2gtk-5/Source/WebCore/platform/graphics/opentype -I/run/build/webkit2gtk-5/Source/WebCore/platform/graphics/transforms -I/run/build/webkit2gtk-5/Source/WebCore/platform/mediacapabilities -I/run/build/webkit2gtk-5/Source/WebCore/platform/mediarecorder -I/run/build/webkit2gtk-5/Source/WebCore/platform/mediastream -I/run/build/webkit2gtk-5/Source/WebCore/platform/mediastream/libwebrtc -I/run/build/webkit2gtk-5/Source/WebCore/platform/mock -I/run/build/webkit2gtk-5/Source/WebCore/platform/mock/mediasource -I/run/build/webkit2gtk-5/Source/WebCore/platform/network -I/run/build/webkit2gtk-5/Source/WebCore/platform/sql -I/run/build/webkit2gtk-5/Source/WebCore/platform/text -I/run/build/webkit2gtk-5/Source/WebCore/platform/xr -I/run/build/webkit2gtk-5/Source/WebCore/plugins -I/run/build/webkit2gtk-5/Source/WebCore/rendering -I/run/build/webkit2gtk-5/Source/WebCore/rendering/line -I/run/build/webkit2gtk-5/Source/WebCore/rendering/mathml -I/run/build/webkit2gtk-5/Source/WebCore/rendering/shapes -I/run/build/webkit2gtk-5/Source/WebCore/rendering/style -I/run/build/webkit2gtk-5/Source/WebCore/rendering/svg -I/run/build/webkit2gtk-5/Source/WebCore/rendering/updating -I/run/build/webkit2gtk-5/Source/WebCore/replay -I/run/build/webkit2gtk-5/Source/WebCore/storage -I/run/build/webkit2gtk-5/Source/WebCore/style -I/run/build/webkit2gtk-5/Source/WebCore/svg -I/run/build/webkit2gtk-5/Source/WebCore/svg/animation -I/run/build/webkit2gtk-5/Source/WebCore/svg/graphics -I/run/build/webkit2gtk-5/Source/WebCore/svg/graphics/filters -I/run/build/webkit2gtk-5/Source/WebCore/svg/properties -I/run/build/webkit2gtk-5/Source/WebCore/websockets -I/run/build/webkit2gtk-5/Source/WebCore/workers -I/run/build/webkit2gtk-5/Source/WebCore/workers/service -I/run/build/webkit2gtk-5/Source/WebCore/workers/service/context -I/run/build/webkit2gtk-5/Source/WebCore/workers/service/server -I/run/build/webkit2gtk-5/Source/WebCore/workers/shared -I/run/build/webkit2gtk-5/Source/WebCore/workers/shared/context -I/run/build/webkit2gtk-5/Source/WebCore/worklets -I/run/build/webkit2gtk-5/Source/WebCore/xml -I/run/build/webkit2gtk-5/Source/WebCore/xml/parser -I/run/build/webkit2gtk-5/Source/WebCore/Modules/gamepad -I/run/build/webkit2gtk-5/Source/WebCore/platform/graphics/lcms -I/run/build/webkit2gtk-5/Source/ThirdParty/xdgmime/src -I/run/build/webkit2gtk-5/Source/WebCore/platform/graphics/cairo -I/run/build/webkit2gtk-5/Source/WebCore/platform/graphics/freetype -I/run/build/webkit2gtk-5/Source/WebCore/platform/graphics/harfbuzz -I/run/build/webkit2gtk-5/Source/WebCore/platform/graphics/harfbuzz/ng -I/run/build/webkit2gtk-5/Source/WebCore/platform/graphics/gstreamer -I/run/build/webkit2gtk-5/Source/WebCore/platform/graphics/gstreamer/mse -I/run/build/webkit2gtk-5/Source/WebCore/platform/graphics/gstreamer/eme -I/run/build/webkit2gtk-5/Source/WebCore/platform/audio/gstreamer -I/run/build/webkit2gtk-5/Source/WebCore/platform/image-decoders -I/run/build/webkit2gtk-5/Source/WebCore/platform/image-decoders/avif -I/run/build/webkit2gtk-5/Source/WebCore/platform/image-decoders/bmp -I/run/build/webkit2gtk-5/Source/WebCore/platform/image-decoders/gif -I/run/build/webkit2gtk-5/Source/WebCore/platform/image-decoders/ico -I/run/build/webkit2gtk-5/Source/WebCore/platform/image-decoders/jpeg -I/run/build/webkit2gtk-5/Source/WebCore/platform/image-decoders/jpeg2000 -I/run/build/webkit2gtk-5/Source/WebCore/platform/image-decoders/jpegxl -I/run/build/webkit2gtk-5/Source/WebCore/platform/image-decoders/png -I/run/build/webkit2gtk-5/Source/WebCore/platform/image-decoders/webp -I/run/build/webkit2gtk-5/Source/WebCore/platform/network/soup -I/run/build/webkit2gtk-5/Source/WebCore/platform/graphics/texmap -I/run/build/webkit2gtk-5/Source/WebCore/platform/graphics/nicosia -I/run/build/webkit2gtk-5/Source/WebCore/page/scrolling/nicosia -I/run/build/webkit2gtk-5/Source/WebCore/platform/graphics/texmap/coordinated -I/run/build/webkit2gtk-5/Source/WebCore/platform/graphics/nicosia/cairo -I/run/build/webkit2gtk-5/Source/WebCore/platform/graphics/nicosia/texmap -I/run/build/webkit2gtk-5/Source/WebCore/accessibility/atk -I/run/build/webkit2gtk-5/Source/WebCore/accessibility/atspi -I/run/build/webkit2gtk-5/Source/WebCore/editing/atk -I/run/build/webkit2gtk-5/Source/WebCore/page/gtk -I/run/build/webkit2gtk-5/Source/WebCore/platform/adwaita -I/run/build/webkit2gtk-5/Source/WebCore/platform/audio/glib -I/run/build/webkit2gtk-5/Source/WebCore/platform/generic -I/run/build/webkit2gtk-5/Source/WebCore/platform/glib -I/run/build/webkit2gtk-5/Source/WebCore/platform/gtk -I/run/build/webkit2gtk-5/Source/WebCore/platform/graphics/egl -I/run/build/webkit2gtk-5/Source/WebCore/platform/graphics/glx -I/run/build/webkit2gtk-5/Source/WebCore/platform/graphics/gbm -I/run/build/webkit2gtk-5/Source/WebCore/platform/graphics/gtk -I/run/build/webkit2gtk-5/Source/WebCore/platform/graphics/opengl -I/run/build/webkit2gtk-5/Source/WebCore/platform/graphics/wayland -I/run/build/webkit2gtk-5/Source/WebCore/platform/graphics/x11 -I/run/build/webkit2gtk-5/Source/WebCore/platform/mediastream/gtk -I/run/build/webkit2gtk-5/Source/WebCore/platform/mediastream/gstreamer -I/run/build/webkit2gtk-5/Source/WebCore/platform/network/glib -I/run/build/webkit2gtk-5/Source/WebCore/platform/text/gtk -I/run/build/webkit2gtk-5/ANGLE/Headers -I/run/build/webkit2gtk-5/JavaScriptCore/Headers -I/run/build/webkit2gtk-5/JavaScriptCore/PrivateHeaders -I/run/build/webkit2gtk-5/JavaScriptCoreGLib/Headers -I/run/build/webkit2gtk-5/JavaScriptCoreGLib/DerivedSources -I/run/build/webkit2gtk-5/WTF/Headers -I/run/build/webkit2gtk-5/bmalloc/Headers -I/run/build/webkit2gtk-5/PAL/Headers -isystem /usr/include/gstreamer-1.0 -isystem /usr/include/glib-2.0 -isystem /usr/lib/aarch64-linux-gnu/glib-2.0/include -isystem /usr/include/sysprof-4 -isystem /usr/include/orc-0.4 -isystem /usr/lib/aarch64-linux-gnu/gstreamer-1.0/include -isystem /usr/include/libsoup-3.0 -isystem /usr/include/atk-1.0 -isystem /usr/include/enchant-2 -isystem /usr/include/gio-unix-2.0 -isystem /usr/include/libmount -isystem /usr/include/blkid -isystem /usr/include/libsecret-1 -isystem /usr/include/libxml2 -isystem /usr/include/cairo -isystem /usr/include/freetype2 -isystem /usr/include/harfbuzz -isystem /usr/include/openjpeg-2.4 -isystem /usr/include/gtk-4.0 -isystem /usr/include/pango-1.0 -isystem /usr/include/libpng16 -isystem /usr/include/fribidi -isystem /usr/include/pixman-1 -isystem /usr/include/gdk-pixbuf-2.0 -isystem /usr/include/graphene-1.0 -isystem /usr/lib/aarch64-linux-gnu/graphene-1.0/include -isystem /usr/include/wpe-1.0 -fdiagnostics-color=always -Wextra -Wall -pipe -Wno-expansion-to-defined -Wno-odr -Wno-stringop-overread -Wno-nonnull -Wno-array-bounds -Wno-noexcept-type -Wno-psabi -Wno-misleading-indentation -Wno-maybe-uninitialized -Wwrite-strings -Wundef -Wpointer-arith -Wmissing-format-attribute -Wformat-security -Wcast-align -Wno-tautological-compare -O2 -g -pipe -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer -fno-strict-aliasing -fno-exceptions -fno-rtti -O3 -DNDEBUG -fPIC -fvisibility=hidden -pthread -I/usr/include/aarch64-linux-gnu -std=c++20 -MD -MT Source/WebCore/CMakeFiles/WebCore.dir/__/__/WebCore/DerivedSources/unified-sources/UnifiedSource-3a52ce78-119.cpp.o -MF Source/WebCore/CMakeFiles/WebCore.dir/__/__/WebCore/DerivedSources/unified-sources/UnifiedSource-3a52ce78-119.cpp.o.d -o Source/WebCore/CMakeFiles/WebCore.dir/__/__/WebCore/DerivedSources/unified-sources/UnifiedSource-3a52ce78-119.cpp.o -c /run/build/webkit2gtk-5/WebCore/DerivedSources/unified-sources/UnifiedSource-3a52ce78-119.cpp
{standard input}: Assembler messages:
{standard input}:184788: Warning: end of file not at end of a line; newline inserted
{standard input}:184789: Error: operand 1 must be a floating-point register -- `stp'
{standard input}: Error: open CFI at the end of file; missing .cfi_endproc directive
c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
```

If it doesn't ring any bell (it doesn't to me), target architectures can be limited: https://github.com/flathub/flathub/wiki/App-Maintenance#only-build-on-x86_64= Would be the best to fix it, but it is what it is.

**Comment by hfiguiere** [comment]
Look like it fails to build a gigantic "unified" file in WebKit, and the compiler get *killed*. But I thought WebKit was actually properly maintained on aarch64.

Is the builder runing out of memory?

**Comment by johannesjh** [comment]
This might well be the case - on my local machine, 16gb were not enough to build webkitgtk. 

Note, I included a custom build of webkitgtk because I need a version that works with GTK4. Same problem and same solution as in gfeeds, compare this [blogpost about gfeeds needing a custom webkit build for gtk4 compatibility](https://gabmus.org/posts/gtk4_libadwaita_and_the_new_feeds/) and [gfeeds's flatpak manifest which includes a custom build of webkitgtk](https://github.com/flathub/org.gabmus.gfeeds). The flathub buildbot website shows that [gfeeds is built successfully](https://buildbot.flathub.org/#/apps/org.gabmus.gfeeds) by flathub's bot.

**Comment by hfiguiere** [comment]
disabling parallel build for the module might be a way to go.

Let's try again.

bot, build org.gnome.gitlab.johannesjh.favagtk

**Comment by johannesjh** [comment]
Note, I am currently experimenting with a new build setup to fix the arm build. I hope to be able to provide an update soon. I used pip-compile-multi to segment python dependencies into multiple groups:

* I build and install webkitgtk from source packages, as in the gfeeds app
* I build and install scikit-learn and its transitive dependencies from source packages, as in the openpaperwork app
* All other python dependencies should hopefully install in a straight-forward way using flatpak-pip-generator

...I'll keep you updated. Thank you for your thoughts and help!

**Comment by hfiguiere** [comment]
this time the webkit build worked. The failure in aarch64 is because of the binary wheels.

**Comment by johannesjh** [comment]
alright, scikit-learn is now also installed from source packages. the build takes a _long_ time now, but building from source seems to be the preferred way to go. The build has worked locally on intel, and I hope it will also work on ARM architectures. If I may say so, 

bot, build org.gnome.gitlab.johannesjh.favagtk

thx

**Comment by johannesjh** [comment]
Interestingly, the ` Validate external-data-checker` step failed with the following error message:
**ManifestFileTooLarge: Manifest file size 116.8 KiB exceeds 100.0 KiB**

```
INFO    src.manifest: Loading module from requirements/python3-main.json
Traceback (most recent call last):
  File "/app/bin/flatpak-external-data-checker", line 30, in <module>
    main()
  File "/app/lib/fedc/src/main.py", line 457, in main
    outdated_num, errors_num, updated = asyncio.run(run_with_args(args))
  File "/usr/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/app/lib/fedc/src/main.py", line 418, in run_with_args
    manifest_checker = manifest.ManifestChecker(args.manifest, options)
  File "/app/lib/fedc/src/manifest.py", line 130, in __init__
    self._collect_external_data()
  File "/app/lib/fedc/src/manifest.py", line 180, in _collect_external_data
    self._collect_module_data(self._root_manifest_path, module)
  File "/app/lib/fedc/src/manifest.py", line 222, in _collect_module_data
    ext_module = self._read_manifest(ext_module_path)
  File "/app/lib/fedc/src/manifest.py", line 159, in _read_manifest
    raise ManifestFileTooLarge(
src.lib.errors.ManifestFileTooLarge: Manifest file size 116.8 KiB exceeds 100.0 KiB: /srv/buildbot/worker/build-x86_64-2/build/requirements/python3-main.json
```

...I can confirm that the manifest file is indeed that large. It would make sense in my opinion to increase the 100KB limit.

**Comment by johannesjh** [comment]
The ARM build failed because it ran into a timeout. It was killed when it was almost done.

I feel like I am running in circles. I refactored the build process to build from sources, as a recommended practice in flathub and in order to support the ARM architecture. Obviously the build now uses more resources... So we now have a failing build because it takes too much time.

Any ideas how we can proceed? 

* Is it possible to allow a longer build time?
* One possible workaround would be to have pip resolve python packages dynamically during the build. This would be very easy because pip automatically selects the right binary packages for each platform. Note however: This would require network access during the build, and python dependencies would not be frozen and hashed. Is this forbidden on flathub, right?
* Another possible workaround regarding how I could structure the application: I could move the add-ons that use machine learning into a flatpak extension, this would effectively double the amount of available build time. 
* Another possible workaround: I could try rewriting the flatpak-pip-generator to generate separate build modules, one for Intel wheels and one for ARM wheels. Not sure if and how this is actually possible (would I have to run flatpak-pip-generator inside an ARM vm). Not sure if this is even wanted, because it would effectively mean that I'd be installing binary packages instead of building from sources.

**Comment by hfiguiere** [comment]
The build failure is an infra problem.

**Comment by hfiguiere** [comment]
bot, build

**Comment by hfiguiere** [comment]
bot, build org.gnome.gitlab.johannesjh.favagtk

**Comment by johannesjh** [comment]
> The build failure is an infra problem.

Oh alright, thank you for looking into this and for giving it another try!

**Comment by johannesjh** [comment]
Thank you, I appreciate your support in getting the app published!

**Pull request closed without merging**